### PR TITLE
v1.1.0: Upstream merge, show address, prettier coins

### PR DESCRIPTION
**Pull request by notatestuser**
**CHANGES**
* Merged with upstream ZondaX/ledger-cosmos-app ~v1.3.2
  * User has to "preview" the tx before signing now
  * Added new ins: `INS_GET_ADDR_SECP256K1`, `INS_SHOW_ADDR_SECP256K1`
  * Memory usage improvements
* Applied decimal formatting for order fields and transfer inputs/outputs
* Applied pretty printing of long json values (i.e. coins)
* Reduced max bip32 depth to 5 and checking that all but the last "node" match as expected: `m/44'/714'/0'/0/...`
* Validating that bech32 hrp is either `tbnb` or `bnb` for "show address"
* "Show address" main menu option can now scroll through address indexes
* Some usability related UI changes
* **Security:** The bip32 path is "locked in" when the user _sees_ the address (either via getting public key, getting the address or showing the address on the device). Trying to sign with a different path (i.e. during a MitM attack) will not work and will instead return an error from the device.

Test cases for new features are in a javascript-sdk branch.

### CODE REVIEW

**Review by darren-liu — DISMISSED** [review]

**Review by darren-liu — APPROVED** [review]

File: src/app_main.c [hunk 1]
```diff
@@ -108,23 +121,103 @@ void app_init() {
...
+          0,               // change (no change addresses for now)
+          placeholder,     // address index
+    };
+    if (depth != 5) {
+        return 0;
+    }
+    if (sizeof(expected) / 4 != depth) {
+        return 0;
+    }
+    for (unsigned int i = 0; i < depth; i++) {
```
Comment (HaoyangLiu) [inline src/app_main.c:171]: How about change `unsigned int` to `uint8_t`, thus we can reduce 3 bytes memory cost. This is a tiny change.

File: src/app_main.c [hunk 2]
```diff
@@ -108,23 +121,103 @@ void app_init() {
...
+    uint32_t placeholder = 0xFFFFFFFF;
+    uint32_t expected[] = {
+         44 | 0x80000000,  // purpose
+        714 | 0x80000000,  // coin type (chain ID)
+          0 | 0x80000000,  // account
+          0,               // change (no change addresses for now)
+          placeholder,     // address index
+    };
+    if (depth != 5) {
+        return 0;
```
Comment (HaoyangLiu) [inline src/app_main.c:166]: Return `true` or `false` instead of `1` or `0`

File: src/app_main.c [hunk 3]
```diff
@@ -108,23 +121,103 @@ void app_init() {
...
+        if (expected[i] == placeholder) continue;
+        if (path[i] != expected[i]) return 0;
+    }
+    return 1;
+}
+
+void set_hrp(char *hrp) {
+    strcpy(bech32_hrp, hrp);
+    bech32_hrp_len = strlen(bech32_hrp);
+}
```
Comment (HaoyangLiu) [inline src/app_main.c:181]: Maybe we should check the string length of hrp against `MAX_BECH32_HRP_LEN`

File: src/app_main.c (hunk 2 above)
Comment (notatestuser) [reply src/app_main.c:166]: Same thing

File: src/app_main.c (hunk 3 above)
Comment (notatestuser) [reply src/app_main.c:181]: This is only used with [hardcoded string consts](https://github.com/binance-chain/ledger-app-binance/blob/d1aebe6240759ae2c8dc75c366ed92612bdd652e/src/view.c#L106)`"bnb"` and `"tbnb"`.

`extract_hrp` and `validate_bnc_hrp` are used for user provided hrp validation.

File: src/app_main.c (hunk 1 above)
Comment (notatestuser) [reply src/app_main.c:171]: Changed in 153df1b52042ec4e47d3bbd8cab8f7754e463bb7

**Pull request merged**